internal/controller/controllers/agentserviceconfig_controller_test.go

```diff
@@ -0,0 +1,189 @@
+package controllers
+
+import (
+	"context"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	routev1 "github.com/openshift/api/route/v1"
+	aiv1beta1 "github.com/openshift/assisted-service/internal/controller/api/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	fakeclient "sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+const (
+	testName                         = "agent"
+	testAgentServiceConfigKind       = "testKind"
+	testAgentServiceConfigAPIVersion = "testAPIVersion"
+)
+
+func newTestReconciler(initObjs ...runtime.Object) *AgentServiceConfigReconciler {
+	c := fakeclient.NewFakeClientWithScheme(scheme.Scheme, initObjs...)
+	return &AgentServiceConfigReconciler{
+		Client:    c,
+		Scheme:    scheme.Scheme,
+		Log:       ctrl.Log.WithName("testLog"),
+		Namespace: testNamespace,
+	}
+}
+
+var _ = Describe("agentserviceconfig_controller reconcile", func() {
```

Comment:
In the future, something like this should really be done in a separate PR.

Reply:
I will remove.